### PR TITLE
[Proxy] Use the same service for exploring extensions and for installing them

### PR DESCRIPTION
**Pull request by fazouane-marouane**
Hi,

This is in relation to https://github.com/Microsoft/vscode/issues/17574, https://github.com/Microsoft/vscode/issues/17397, https://github.com/Microsoft/vscode/issues/17360.

Like I said in those issues, I remarked that looking for extensions on the marketplace and installing them don't use the same "piece of code". And indeed, listing extensions and their properties is done by the main process which has access to ExtensionManagementService. The latter doesn't seem to care about the config "http.proxy", but can nonetheless use the default system proxy. So far, so good.

However, when "install" is pressed on the screen, the IExtensionManagementService is resolved as ExtensionManagementChannelClient which will call via ipc the Shared process which will finally call ExtensionManagementService. What's astonishing is that the same piece of code now cares about "http.proxy" and when it's not set, it cannot use the default system proxy.

**TL;DR**
This PR aims at fixing the issues with http proxy or otherwise be a starting point 😁 
To test that the behaviours of searchs & installs are different you can add the following user setting: 

```json
{
    "http.proxy": "http://dummy.proxy:8080"
}
```

What do you think? Is the fix proposed here the right thing to do or does it hide another bug?

### CODE REVIEW

**Comment by fazouane-marouane** [comment]
Ok. So it seems that there are two classes names RequestService.
The first one is defined in 'vs/platform/request/node/requestService' and can be run on a node process and is used in all the processes except the shell.
The shell uses the other one, which is defined in 'vs/platform/request/electron-browser/requestService'. What's different about this implementation is that it uses XMLHttpRequest(so it can't be used in the shared process since it's running on top of node and not electron) and completely ignores proxyagent which we pass inside a IRequestOptions.

So this leaves two possible fixes:
1. Query extensions & install them from the same process using the same service, but totally ignore the "proxy.http". Which is cool since VSCode will still behave like any chrome instance and use the default system proxy (*And that's what this pull request does*)
2. Always use RequestService from 'vs/platform/request/node/requestService'. But we'll always need to specify the "http.proxy" settings (the least favorite solution)

What do you think?

**Comment by joaomoreno** [comment]
Closing this PR as there should only be one instance of `ExtensionManagementService` at any time.

I will implement using Chrome's stack for network requests.
